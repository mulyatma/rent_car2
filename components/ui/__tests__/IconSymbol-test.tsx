import React from 'react';
import renderer from 'react-test-renderer';
import MaterialIcons from '@expo/vector-icons/MaterialIcons';
import { IconSymbol } from '../IconSymbol';

describe('IconSymbol Component', () => {
  it('renders correctly with default props', () => {
    const component = renderer.create(
      <IconSymbol name="house.fill" color="black" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct MaterialIcon based on the mapping', () => {
    const component = renderer.create(
      <IconSymbol name="car.fill" color="red" size={30} />
    );
    const instance = component.root;

    // Find the MaterialIcons component
    const materialIcon = instance.findByType(MaterialIcons);

    expect(materialIcon.props.name).toBe('directions-car'); // Verify the mapped name
    expect(materialIcon.props.color).toBe('red'); // Verify the color
    expect(materialIcon.props.size).toBe(30); // Verify the size
  });

  it('applies the style prop correctly', () => {
    const style = { margin: 10 };
    const component = renderer.create(
      <IconSymbol name="paperplane.fill" color="blue" size={24} style={style} />
    );
    const instance = component.root;

    const materialIcon = instance.findByType(MaterialIcons);
    expect(materialIcon.props.style).toBe(style); // Verify the style prop
  });

  it('throws an error for an invalid icon name', () => {
    // Using a try-catch to handle the error and check its message
    expect(() =>
      renderer.create(<IconSymbol name="invalid.icon" color="black" />)
    ).toThrowError('undefined is not an object (evaluating \'MAPPING[name]\')');
  });
});
