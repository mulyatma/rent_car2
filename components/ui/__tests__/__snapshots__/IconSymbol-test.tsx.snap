// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconSymbol Component renders correctly with default props 1`] = `
<ViewManagerAdapter_SymbolModule
  animated={false}
  colors={[]}
  name="house.fill"
  resizeMode="scaleAspectFit"
  style={
    [
      {
        "height": 24,
        "width": 24,
      },
      [
        {
          "height": 24,
          "width": 24,
        },
        undefined,
      ],
    ]
  }
  tintColor="black"
  type="monochrome"
  weight="regular"
/>
`;
